docs/how-to-build-with-ddn/with-bigquery.mdx

```diff
@@ -0,0 +1,442 @@
+        "name": "Alice",
+        "age": "25"
+      },
+      {
+        "userId": "3",
+        "name": "Charlie",
+        "age": "35"
+      },
+      {
+        "userId": "2",
+        "name": "Bob",
+        "age": "30"
+      }
+    ]
+  }
+}
+```
+
+### Step 12. Iterate on your BigQuery schema
+
+Add a new table to your BigQuery dataset with the name `posts`. You can use the following schema text definition to
+create the table in the console:
+
+```json title="Add a new posts table to your BigQuery dataset:"
+[
+  {
+    "name": "user_id",
+    "type": "INTEGER",
+    "mode": "REQUIRED"
+  },
+  {
+    "name": "post_id",
+    "type": "INTEGER",
+    "mode": "REQUIRED"
+  },
+  {
+    "name": "title",
+    "type": "STRING",
+    "mode": "NULLABLE",
+    "maxLength": "255"
+  },
+  {
+    "name": "content",
+    "type": "STRING",
+    "mode": "NULLABLE"
+  }
+]
+```
+
+Once created, you can click on the table and the "Query" button to seed the table with some data:
+
+```SQL title="Seed the new table:"
+INSERT INTO hasura_demo.posts (user_id, post_id, title, content) VALUES
+  (1, 1, 'My First Post', 'This is the first post for Alice.'),
+  (1, 2, 'Another Post', 'Alice writes again!'),
+  (2, 3, 'Bobs Post', 'Bob shares his thoughts.'),
+  (3, 4, 'Hello World', 'Charlie joins the conversation.');
+```
+
+### Step 13. Refresh your metadata and rebuild your project
+
+:::tip
+
+The following steps are necessary each time you make changes to your **source** schema. This includes, adding,
+modifying, or dropping tables.
+
+:::
+
+#### Step 13.1. Re-introspect your data source
+
+```sh title="Run the introspection command again:"
+ddn connector introspect my_bigquery
+```
+
+In `app/connector/my_bigquery/configuration.json`, you'll see schema updated to include operations for the `posts`
+table.
+
+#### Step 13.2. Update your metadata
+
+```sh title="Add the posts model:"
+ddn model add my_bigquery posts
+```
+
+In `app/metadata/my_bigquery.hml`, you'll see `posts` present in the metadata.
+
+### Step 14. Create a Relationship in your Hasura metadata
+
+Let's also now create a Relationship in our Hasura metdata for between the `users` and `posts` tables.
+
+Manually add the following to the bottom of your `app/metadata/Posts.hml` file. The
+[Hasura VS Code LSP extension](https://marketplace.visualstudio.com/items?itemName=HasuraHQ.hasura) can help you with
+this.
+
+```yaml title="Add a new relationship to your Hasura metadata:"
+---
+kind: Relationship
+version: v1
+definition:
+  name: user
+  sourceType: Posts
+  target:
+    model:
+      name: Users
+      relationshipType: Object
+  mapping:
+    - source:
+        fieldPath:
+          - fieldName: userId
+      target:
+        modelField:
+          - fieldName: userId
+```
+
+Since this is added directly to the metadata, you don't need to run any commands to add it from the introspection of the
+connector. By building your supergraph you will make it available in your API.
```

Comment:
Would we consider adding this to other guides that also have hand-written `relationships`? If so, let's create a clean-up task to add these across for consistency.

Reply:
Created in project updates in Linear 👍